File: wheeled_vehicle_models/18dof/pydof18_demos/HMMWV/demo_hmmwv_hi.py
# Demo for hmmwv simulation using the half implicit integrator in python

import numpy as np
import pydof18 as dof18
import sys

# Input file
driver_file = "../../data/input/" + str(sys.argv[1]) + ".txt"

# Vehicle specification
vehParamsJson = "../../data/json/HMMWV/vehicle.json"
tireParamsJson = "../../data/json/HMMWV/tmeasy.json"


# Construct the half Implicit solver
solver = dof18.d18SolverHalfImplicit()
solver.Construct(vehParamsJson, tireParamsJson, driver_file)


# Set the simulation time step
solver.SetTimeStep(1e-3)

# Iniitial conditions
veh_st = dof18.VehicleState()
tirelf_st = dof18.TMeasyState()
tirerf_st = dof18.TMeasyState()
tirelr_st = dof18.TMeasyState()
tirerr_st = dof18.TMeasyState()
solver.Initialize(veh_st, tirelf_st, tirerf_st, tirelr_st, tirerr_st)

# Enable simulation output at 100 Hz
solver.SetOutput("../../data/output/" +
                 str(sys.argv[1]) + "_hmmwv18HiPy.csv", 100)

# Solve
solver.Solve()

File: wheeled_vehicle_models/18dof/demos/HMMWV/demo_hmmwv_hi.cpp
// =============================================================================
// Authors: Huzaifa Unjhawala
// =============================================================================
//
// A HMMWV vehicle is defined using example JSON files.
// In addition, the user is required to provide a driver input file. 
// Example driver input files are provided in the data/input folder.
// The Half-Implicit solver is then Constructed, Initialized and solved. Data at the specified
// output frequency is written to the specified output file.
//
// =============================================================================

#include <numeric>
#include <algorithm>
#include <iterator>

#include "dof18_halfImplicit.h"

using namespace d18;

int main(int argc, char** argv) {
    // Driver inputs
    std::string driver_file = "../data/input/" + std::string(argv[1]) + ".txt";

    // Vehicle specification
    std::string vehParamsJSON = (char*)"../data/json/HMMWV/vehicle.json";
    std::string tireParamsJSON = (char*)"../data/json/HMMWV/tmeasy.json";

    // Construct the solver
    d18SolverHalfImplicit solver;
    solver.Construct(vehParamsJSON, tireParamsJSON, driver_file);

    // Set solver time step
    solver.SetTimeStep(1e-3);

    // Initialize solver (set initial conditions)
    VehicleState veh_st;
    TMeasyState tirelf_st;
    TMeasyState tirerf_st;
    TMeasyState tirelr_st;
    TMeasyState tirerr_st;
    solver.Initialize(veh_st, tirelf_st, tirerf_st, tirelr_st, tirerr_st);

    // Enable output
    solver.SetOutput("../data/output/" + std::string(argv[1]) + "_hmmwv18Hi.csv", 100);
    // Solve
    solver.Solve();
}